umbraco/Umbraco.CMS.Backoffice
# Make Grid List Create Button Respect the Translation of a document type

This PR is a fix for Both : 

Fixes https://github.com/umbraco/Umbraco-CMS/issues/17233
Fixes https://github.com/umbraco/Umbraco-CMS/issues/17234


This PR simply checks if the document type label starts with '#' which means that the doctype label is translatable
and then removes the '#' and applies the translations 
Before :
<img width="716" alt="image" src="https://github.com/user-attachments/assets/44496468-72f7-4915-bbf1-8592f9b6fa11">
<img width="851" alt="image" src="https://github.com/user-attachments/assets/8af680cc-c964-4cec-9d17-474f1d1216a5">


After:
<img width="701" alt="image" src="https://github.com/user-attachments/assets/828e14da-5803-4cb0-8022-784243ee95e9">
<img width="496" alt="image" src="https://github.com/user-attachments/assets/a288aca4-9a20-42fc-89c9-23a256da69be">





## Motivation and context



## How to test?

1-Create a block list for any doc type with translations
2- Go to block type and Check the CreateButtonLabel
3-you will notice that the label is translated

## Screenshots (if appropriate)
Before :
<img width="716" alt="image" src="https://github.com/user-attachments/assets/44496468-72f7-4915-bbf1-8592f9b6fa11">
<img width="851" alt="image" src="https://github.com/user-attachments/assets/8af680cc-c964-4cec-9d17-474f1d1216a5">

After:
<img width="701" alt="image" src="https://github.com/user-attachments/assets/828e14da-5803-4cb0-8022-784243ee95e9">

<img width="496" alt="image" src="https://github.com/user-attachments/assets/a288aca4-9a20-42fc-89c9-23a256da69be">


## Checklist



- [] I have added tests to cover my changes.
